### ISSUE TITLE
incompatible_use_jdk10_as_host_javabase: makes JDK 10 the default --host_javabase for remote jdk

### ISSUE THREAD

**Issue by cushon** [bazelbuild/bazel#6661]
The option `--incompatible_use_jdk10_as_host_javabase` makes JDK 10 the default [`--host_javabase`](https://docs.bazel.build/versions/master/user-manual.html#flag--host_javabase).

See also `--incompatible_use_remotejdk_as_host_javabase` (#6656).

# Migration notes

If you set the flag, Bazel will use JDK 10 as the default [`--host_javabase`](https://docs.bazel.build/versions/master/user-manual.html#flag--host_javabase).

For more information about JDK 10, see [the release notes](https://www.oracle.com/technetwork/java/javase/10-relnote-issues-4108729.html).

**Comment by dslomov** [comment]
Where does it get JDK10? Do you mean it downloads JDK10 as a remote JDK?

**Comment by cushon** [comment]
> Do you mean it downloads JDK10 as a remote JDK?

Right, see also #6656.

As discussed with @lberki I think this might be slated for 0.22 instead.

**Comment by laurentlb** [comment]
The flag was not flipped. Is it planned for 0.22 instead?

**Comment by cushon** [comment]
Yep, that's still the plan.

**Comment by laurentlb** [comment]
@dslomov Please create the label "breaking-change-0.22" (and others).

**Comment by meteorcloudy** [comment]
Wait, looks like this change will breaking building itself, see https://buildkite.com/bazel/bazel-at-release-plus-incompatible-flags/builds/50#38e92ffb-452e-4235-8eb8-e4119857b1bd
I discovered this after adding Bazel itself as a downstream project.

**Comment by cushon** [comment]
I can't reproduce the failure; `bazel build --incompatible_use_jdk10_as_host_javabase //src:bazel` works for me.

The interesting part of the log in https://github.com/bazelbuild/bazel/issues/6661#issuecomment-451147790 seems to be:

```
ERROR: Skipping '//src:bazel_jdk_minimal': no such target '//src:bazel_jdk_minimal': target 'bazel_jdk_minimal' not declared in package 'src' defined by /var/lib/buildkite-agent/builds/bk-worker-ubuntu1404-java8-k24w/bazel-downstream-projects/bazel/src/BUILD
```

@meisterT any ideas? Is this an interaction between the changes to minify the embedded JDK and `--incompatible_use_jdk10_as_host_javabase`?

**Comment by meisterT** [comment]
That just must have been a weird state of the repository. The host javabase is completely unrelated to that error. I tried manually and it didn't fail, so I think we're fine.

**Comment by philwo** [comment]
I verified that this flag was flipped and thus will be included in the Bazel 0.23.0 release.
